SDLnetsys.h

```diff
@@ -73,6 +73,10 @@ typedef int socklen_t;
 typedef Uint32 socklen_t;
 #endif
 
+#if defined(__USE_W32_SOCKETS) && !defined(socklen_t)
+typedef SOCKET socklen_t;
```

Comment:
For windows, socklen_t is a `signed int` (whereas `SOCKET` is unsigned), and is a winsock2 thing.
Below is an adaptation from one of my projects:

```c
/* on windows, socklen_t is to be a winsock2 thing */
#if defined(__USE_W32_SOCKETS) && !defined(IP_MSFILTER_SIZE)
typedef int socklen_t;
#endif
```